Template/Trigger/ElementVisibilityTrigger.web.js

```diff
@@ -310,10 +294,86 @@
             return false;
         }
 
+        function setMutationObserver(triggerEvent) {
+            return function () {
+                if (observeDomChanges && isMutationObserverSupported) {
+                    var targetNode = documentAlias.body;
+                    var config = {attributes: true, childList: true, subtree: true};
+                    observerMutation = new MutationObserver(function (mutationsList) {
+                        mutationObserverCallback(mutationsList, triggerEvent);
+                    });
+                    observerMutation.observe(targetNode, config);
+                }
+            };
+        }
+
+        function mutationObserverCallback(mutationsList, triggerEvent) {
+            for (var index in mutationsList) {
+                var mutation = mutationsList[index];
+                if (mutation.type === 'childList' || mutation.type === 'attributes') {
+                    var addedNodes = mutation.addedNodes;
+                    addedNodes.forEach(function (node) {
+                        TagManager.dom.bySelector(selectors).forEach(function (element) {
+                            if (node.contains(element)) {
+                                if (blockTrigger || (onlyOncePerElement && isNodeEventTriggered(element))) {
+                                    return;
+                                }
+
+                                if (!isNodeInViewport(element) && observerIntersection) {
+                                    observerIntersection.observe(element);
```

Comment:
I haven't looked too deep yet but if this callback is triggered multiple times, and the element is not in the viewport, does this mean we start observing the element every time meaning we call `observerIntersection.observe(element)` several times for the same element under circumstances?

Reply:
Yes nice catch!
I added a fix to trigger only once